buildsystem/spack/load_spack.sh

```diff
@@ -2,79 +2,69 @@
+	echo "Try one of the following platforms: "
+	echo $(ls -d ./buildsystem/spack/*/ | tr '\n' '\0' | xargs -0 -n 1 basename)
+	return
 fi
 
 base="./buildsystem/spack/${MY_CLUSTER}"
 
 # There needs to be an existing spack.yaml
-if [ ! -f "$base/spack.yaml" ]
-then
-  echo "No spack.yaml in $base/spack.yaml. Add this and try again."
-  return
+if [ ! -f "$base/spack.yaml" ]; then
+	echo "No spack.yaml in $base/spack.yaml. Add this and try again."
+	return
 fi
 
 # We must configure with platform specific environment before loading spack
-if [ ! -f "$base/env.sh" ]
-then
-  echo "No environment (env.sh) script detected in $base."
-  echo "Please add a script that creates a compatible shell environment."
-  echo "You must set the following variables that affect your spack configuration."
-  echo "TODO - add updated description of all spack platform variables" # TODO
-  return
+if [ ! -f "$base/env.sh" ]; then
+	echo "No environment (env.sh) script detected in $base."
+	echo "Please add a script that creates a compatible shell environment."
+	echo "You must set the following variables that affect your spack configuration."
+	echo "TODO - add updated description of all spack platform variables" # TODO
+	return
 fi
 
-
 # Source base platform environment
-source $base/env.sh && \
-
-# Make sure SPACK_INSTALL is set, so we aren't installing somewhere undesired
-SPACK_INSTALL="${SPACK_INSTALL:?SPACK_INSTALL is unset. $base/env.sh should be edited to configure this}" && \
-SPACK_MIRROR="${SPACK_MIRROR:?SPACK_MIRROR is unset. $base/env.sh should be edited to configure this}" && \
-
-# Load spack
-source ./tpl/spack/share/spack/setup-env.sh && \
-
-# Create directory for environment
-SPACKENV=$(pwd)/spack-env-$MY_CLUSTER && \
-mkdir -p $SPACKENV && \
-# Remove old config
-(rm -f $SPACKENV/spack.yaml || true) && \
-(rm -f $SPACKENV/spack.lock || true) && \
+source $base/env.sh &&
 
-# Use a directory based environment, and decorate command line
-spack env create -d $SPACKENV && \
+	# Make sure SPACK_INSTALL is set, so we aren't installing somewhere undesired
+	SPACK_INSTALL="${SPACK_INSTALL:?SPACK_INSTALL is unset. $base/env.sh should be edited to configure this}" &&
+	SPACK_MIRROR="${SPACK_MIRROR:?SPACK_MIRROR is unset. $base/env.sh should be edited to configure this}" &&
 
-# Use git version of config
-cp $base/spack.yaml $SPACKENV && \
-spack env activate -p $SPACKENV && \
+	# Load spack
+	source ./tpl/spack/share/spack/setup-env.sh &&
 
-# Print relevant spack config for sanity check of environment.
-echo "spack configuration will be installed into $SPACK_INSTALL" && \
-mkdir -p $SPACK_INSTALL && \
-mkdir -p $SPACK_CACHE && \
+	# Create directory for environment
+	SPACKENV=$(pwd)/spack-env-$MY_CLUSTER &&
+	mkdir -p $SPACKENV &&
+	# Remove old config
+	(rm -f $SPACKENV/spack.yaml || true) &&
+	(rm -f $SPACKENV/spack.lock || true) &&
 
+	# Use a directory based environment, and decorate command line
+	spack env create -d $SPACKENV $base/spack.yaml &&
```

Comment:
This is the main change.... No longer are we running `spack env create` before a `cp` command, we just do it all in one line using spack native commands.